### PR TITLE
Reduce settings manager complexity by loading sections via DI

### PR DESCRIPTION
**Pull request by ChristophWurst**
For https://github.com/nextcloud/server/issues/11381 I need to add some more section to the server's default settings sections and therefore I stumbled over this massively overcomplicated class that build the settings sections for both personal and admin settings.

IMO it's a lot easier and cleaner to load these external classes via the DI container. This way changes to them don't have to be reflected in the settings manager (and they shouldn't -> single responsibility principle).

### CODE REVIEW

File: lib/private/Settings/Manager.php [hunk 1]
```diff
@@ -29,96 +29,45 @@
...
+		IL10N $l10n,
 		IURLGenerator $url,
-		AccountManager $accountManager,
-		IGroupManager $groupManager,
-		IFactory $l10nFactory,
-		IAppManager $appManager
+		IServerContainer $container
 	) {
 		$this->log = $log;
 		$this->dbc = $dbc;
```
Comment (nickvergessen) [inline lib/private/Settings/Manager.php]: :see_no_evil: a new participant in the fight of `$db` vs. `$connection` vs. `$dbConn`?

Comment (nickvergessen) [reply lib/private/Settings/Manager.php]: Ah, it's not used now anymore, so you can delete it too

**Review by blizzz — APPROVED** [review]

File: lib/private/Settings/Manager.php (hunk 1 above)
Comment (ChristophWurst) [reply lib/private/Settings/Manager.php]: Good catch. Removed.

**Review by rullzer — APPROVED** [review]
:rocket:

**Comment by ChristophWurst** [comment]
CI failure is unrelated.

**Comment by ChristophWurst** [comment]
> [Christoph's Tasks](/orgs/nextcloud/projects/4) automation moved this from **IN PROGRESS (max 3 PRs)** to **TO INTEGRATE**

Isn't it awesome how the automation knows two reviews are required and once done moves this ticket on the project board? :rocket:

**Pull request merged**